### DISCUSSION TITLE
Milvus Upsert Question: How do we specify the field name during upsert?

### DISCUSSION THREAD

**Discussion by SaratM34** [milvus-io/milvus#30084] [Q&A and General discussion] [answered]
It was mentioned [here](https://milvus.io/docs/upsert_entities.md) that: 

Upserting is a combination of insert and delete operations. In the context of a Milvus vector database, an upsert is a data-level operation that will overwrite an existing entity _**if a specified field already exists in a collection**_, and insert a new entity if the specified value doesn’t already exist.

How do we specify the field name during upsert? I don't see any parameter to provide field name during collection.upsert().

**Comment by yhmo** [comment]
Not possible to specify the field name in current versions.
This is a known requirement, a feature issue is created: https://github.com/milvus-io/milvus/issues/29735
The solution is not completely defined, TBD. If you have any ideas, add your comments in the feature issue.

**Reply by SaratM34** [reply]
thanks for you reply @yhmo . Could you please explain how upsert work currently if no field is specified? Will upsert overwrite existing record in collection only if values in all fields matches with new record we are inserting? or Will it check only primary key field?

**Reply by yhmo** [reply, marked as answer]
Only check the primary key value. If the primary key value already exists, the upsert() will update this row with the input values. Otherwise, the upsert() performs an insert() action to add a new row.
